bin/circleci-docker-build.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+set -ex
+
+DOCKER_CACHE_PATH=~/docker
+DOCKER_CACHE_FILE="${DOCKER_CACHE_PATH}/image.tgz"
+
+mkdir -p $DOCKER_CACHE_PATH
+make clean
+
+if [[ -f $DOCKER_CACHE_FILE ]]; then
+  gunzip -c "$DOCKER_CACHE_FILE" | docker load;
+fi
+
+if [[ -f $DOCKER_CACHE_PATH/bedrock.db ]]; then
+  mv $DOCKER_CACHE_PATH/bedrock.db ./
+fi
+
+LOCALES_TAR_FILE="$DOCKER_CACHE_PATH/locales.tgz"
+if [[ -f "$LOCALES_TAR_FILE" ]]; then
+  tar xzf "$LOCALES_TAR_FILE"
+  rm -f "$LOCALES_TAR_FILE"
+fi
+
+MFSA_TAR_FILE="$DOCKER_CACHE_PATH/mfsa_repo.tgz"
+if [[ -f "$MFSA_TAR_FILE" ]]; then
+  tar xzf "$MFSA_TAR_FILE"
+  rm -f "$MFSA_TAR_FILE"
+fi
+
+PD_TAR_FILE="$DOCKER_CACHE_PATH/pd_files.tgz"
+if [[ -f "$PD_TAR_FILE" ]]; then
+  tar xzf "$PD_TAR_FILE"
+  rm -f "$PD_TAR_FILE"
+fi
+
+if [[ "$CIRCLE_BRANCH" == demo__* ]]; then
```

Comment:
Maybe out of the scope of this PR but it seems that all PRs get advantage of cached locales, database, mfsa and pd but only the ones with branch name matching `demo__*` replace the files.

Maybe the files should get deleted after extracting, only when branch name matches `demo__*`?